02-peer-protocol.md

```diff
@@ -2721,12 +3356,40 @@ The sending node:
   - if it has sent `commitment_signed` for an interactive transaction construction but
     it has not received `tx_signatures`:
     - MUST set `next_funding_txid` to the txid of that interactive transaction.
+    - if it has not received `commitment_signed` for that interactive transaction:
+      - MUST set `next_commitment_number` to the commitment number of the `commitment_signed` it sent.
   - otherwise:
     - MUST NOT set `next_funding_txid`.
+  - if `option_splice` was negotiated:
+    - MUST set `your_last_funding_locked` to the txid of the last `splice_locked` it received.
+    - if it never received `splice_locked` for any transaction, but it received `channel_ready`:
+      - MUST set `your_last_funding_locked` to the txid of the channel funding transaction.
+    - otherwise (it has never received `channel_ready` or `splice_locked`):
+      - MUST NOT set `your_last_funding_locked`.
+    - if a splice transaction reached acceptable depth while disconnected:
+      - MUST set `my_current_funding_locked` to the txid of the latest such transaction.
+      - MUST send `splice_locked` for that transaction after exchanging `channel_reestablish`.
+    - otherwise:
+      - MUST set `my_current_funding_locked` to the txid of the last `splice_locked` it sent.
+      - if it never sent `splice_locked` for any transaction, but it sent `channel_ready`:
+        - MUST set `my_current_funding_locked` to the txid of the channel funding transaction.
+      - otherwise (it has never sent `channel_ready` or `splice_locked`):
+        - MUST NOT set `my_current_funding_locked`.
+      - if `my_current_funding_locked` is included:
+        - if `announce_channel` is set for this channel:
+          - if it has not received `announcement_signatures` for that transaction:
+            - MUST retransmit `channel_ready` or `splice_locked` after exchanging `channel_reestablish`.
+        - if it receives `channel_ready` for that transaction after exchanging `channel_reestablish`:
+          - MUST retransmit `channel_ready` in response, if not already sent since reconnecting.
+        - if it receives `splice_locked` for that transaction after exchanging `channel_reestablish`:
+          - MUST retransmit `splice_locked` in response, if not already sent since reconnecting.
```

Comment:
As written these requirements are dependent on receiving other messages. This seems more complicated than it needs to be. Instead, can't the retransmission requirements be entirely within the `channel_reestablish`'s last "A receiving node" section? There's already a requirement there to retransmit `splice_locked`, so the one here seems redundant. We'd just need to add a requirement there for retransmitting `channel_ready`.

Am I missing something?

Reply:
Given the spec already states that an redundant `channel_ready` messages must be ignored, any reason why we don't just do the same with `splice_locked` and always re-send `channel_ready` / `splice_locked` upon connection? Seems that would simplify the requirements quite a bit.